### PR TITLE
SUBMARINE-360. [WEB] Implement the frontend of sign-up page with Angular

### PR DESCRIPTION
**Pull request by kevin85421**
### What is this PR for?
Implement the frontend of the sign-up page with Angular

Validators :
1. Username: (1) empty (2) existed
2. Email: (1) empty (2) existed
3. Password: (1) empty (2) length
4. Re-enter Password: (1) empty (2) must be the same with the Password
5. Sign Up button: the button can be clicked only when the form is valid.

### What type of PR is it?
[Feature]

### Todos
the backend of the sign-up page

### What is the Jira issue?
https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-360

### How should this be tested?
https://travis-ci.org/kevin85421/hadoop-submarine?utm_medium=notification&utm_source=github_status

### Screenshots (if appropriate)
截圖 2020-02-11 下午2 24 53

### Questions:
* Does the licenses files need an update? No
* Are there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-workbench/workbench-web-ng/src/app/pages/user/register/register.component.ts [hunk 1]
```diff
@@ -18,14 +18,86 @@
...
 
 @Component({
   selector: 'submarine-register',
   templateUrl: './register.component.html',
   styleUrls: ['./register.component.scss']
 })
 export class RegisterComponent implements OnInit {
-  constructor() {}
+  validateForm: FormGroup;
+  // mock data: must be removed in the future
```
Comment (liuxunorg) [inline submarine-workbench/workbench-web-ng/src/app/pages/user/register/register.component.ts]: You can remind yourself by adding `TODO(kevin85421): must be removed in the future`, all TODO will be displayed in the `TODO view` in IDEA.

File: submarine-test/e2e/src/test/java/org/apache/submarine/integration/sidebarIT.java [hunk 2]
```diff
@@ -66,7 +68,13 @@ public void sidebarNavigation() throws Exception {
     pollingWait(By.xpath("//span[contains(text(), \"Manager\")]"), MAX_BROWSER_TIMEOUT_SEC).click();
     pollingWait(By.xpath("//a[@href='/workbench/manager/user']"), MAX_BROWSER_TIMEOUT_SEC).click();
     SubmarineITUtils.sleep( 5000, true);
+
+    // Lazy-loading
+	  WebDriverWait wait = new WebDriverWait( driver, 15, 5000);
+	  pollingWait(By.xpath("//a[@href='/workbench/manager/user']"), MAX_BROWSER_TIMEOUT_SEC).click();  
+	  wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//span[@class='ant-breadcrumb-link ng-star-inserted']")));
```
Comment (liuxunorg) [inline submarine-test/e2e/src/test/java/org/apache/submarine/integration/sidebarIT.java]: Wrong code indentation

File: submarine-workbench/workbench-web-ng/src/app/pages/user/register/register.component.ts (hunk 1 above)
Comment (kevin85421) [reply submarine-workbench/workbench-web-ng/src/app/pages/user/register/register.component.ts]: Thank for your review, @liuxunorg 
I have fixed the problem.

File: submarine-test/e2e/src/test/java/org/apache/submarine/integration/sidebarIT.java (hunk 2 above)
Comment (kevin85421) [reply submarine-test/e2e/src/test/java/org/apache/submarine/integration/sidebarIT.java]: Thank for your review, @liuxunorg 
I have fixed the problem.

**Comment by liuxunorg** [comment]
Will merge if no more comments.

**Pull request closed without merging**